Kitodo/src/main/java/org/kitodo/production/forms/validators/ImportConfigurationMappingValidator.java

```diff
@@ -49,41 +51,49 @@ public void validate(FacesContext facesContext, UIComponent uiComponent, DualLis
+                            Helper.getTranslation("importConfig.validation.error.firstMappingFormat",
+                                    mappingFile.getTitle(), metadataFormat, mappingFile.getInputMetadataFormat()), null));
                 }
-                // last mapping file must map to KITODO internal metadata format
-                if (index == mappingFiles.size() - 1
-                        && !Objects.equals(mappingFile.getOutputMetadataFormat(), MetadataFormat.KITODO.name())) {
-                    throw new ValidatorException(new FacesMessage(FacesMessage.SEVERITY_ERROR,
-                            "Last configured mapping file '" + mappingFile.getTitle()
-                                    + "' must map to KITODO internal format (found '"
-                                    + mappingFile.getOutputMetadataFormat() + "' instead)", null));
+                if (index == mappingFiles.size() - 1) {
+                    validateLastMappingFile(mappingFile, prestructuredImport);
                 }
                 // intermediate mapping files output metadata format must match next files input metadata format
                 if (mappingFiles.size() > index + 1 && !Objects.equals(mappingFile.getOutputMetadataFormat(),
                         mappingFiles.get(index + 1).getInputMetadataFormat())) {
-                    throw new ValidatorException(new FacesMessage(FacesMessage.SEVERITY_ERROR, "Mapping file '"
-                            + mappingFile.getTitle() + "' output metadata format must match next mapping file's input metadata format '"
-                            + mappingFiles.get(index + 1).getInputMetadataFormat() + "' (found '"
-                            + mappingFile.getOutputMetadataFormat() + "' instead!)", null));
+                    throw new ValidatorException(new FacesMessage(FacesMessage.SEVERITY_ERROR, Helper.getTranslation(
+                            "importConfig.validation.error.intermediateMappingFormat",
+                            mappingFile.getTitle(), mappingFiles.get(index + 1).getInputMetadataFormat(),
+                            mappingFile.getOutputMetadataFormat()), null));
                 }
             }
         }
     }
+
+    private void validateLastMappingFile(MappingFile mappingFile, Boolean prestructuredImport) {
+        // last mapping file must map to KITODO internal metadata format
+        if (!Objects.equals(mappingFile.getOutputMetadataFormat(), MetadataFormat.KITODO.name())) {
+            throw new ValidatorException(new FacesMessage(FacesMessage.SEVERITY_ERROR, Helper.getTranslation(
+                    "importConfig.validation.error.lastMappingFormat", mappingFile.getTitle(),
+                    mappingFile.getOutputMetadataFormat()), null));
+        }
+        // 'prestructured import' flag of mapping file must be equal to corresponding flag in import configuration
+        if (!Objects.equals(mappingFile.getPrestructuredImport(), prestructuredImport)) {
```

Comment:
Is this a convention to use Object.equals for comparing Booleans? If not I would suggest to check them without Object.equals.

Reply:
Since `Boolean` objects can be `null` - in contrast to the primitive type `boolean` - I prefer using `Objects.equals`, because it includes a null check to prevent null pointer exceptions.

Reply:
Yes, you are right, it is not a primitive type. Take back my comments here. Did not say anything. ;)